### PR TITLE
Add section for keyboard navigation and pane navigator #107

### LINKED ISSUE #107: Update DockManager topic with Keyboard navigation and Pane Navigator features
Add sections for the following new features to DockManager's topic:
- [x] Keyboard Navigation
- [x] Pane Navigator
- [x] allowFloatingPanesResize/ floatingResizable properties
- [x] Reorder tabs in a tab group via drag & drop (#https://github.com/IgniteUI/dock-manager/issues/81)

### PR DESCRIPTION
**Pull request by zhosti**
Closes #107

### CODE REVIEW

File: doc/en/components/dock-manager.md [hunk 1]
```diff
@@ -325,6 +325,47 @@ The Dock Manager component raises events when specific end-user interactions are
...
+ - <kbd>Cmd/Ctrl + Shift + F6</kbd> Focus previous tab in document host
+ - <kbd>Cmd/Ctrl + Arrow left</kbd> Focus previous tab in document host
+ - <kbd>Alt + F3</kbd> Closes tab/document
+
+### Pane Navigator
+
+With the key combinations for the Pane Navigator you can iterate forward and backwards through panes and documents.
+
+ - <kbd>Cmd/Ctrl + F7</kbd> Shows the navigator and iterate panes and documents starting from the first document forward
+ - <kbd>Cmd/Ctrl + F8</kbd> Shows the navigator and iterate panes and documents starting from the first document forward
```
Comment (igdmdimitrov) [inline doc/en/components/dock-manager.md:359]:
```suggestion
 - <kbd>Cmd/Ctrl + F7</kbd> / <kbd>Cmd/Ctrl + F8</kbd> Shows the navigator and iterate panes and documents starting from the first document forward
```

Where possible, combine shortcuts that have the same function.

File: doc/en/components/dock-manager.md [hunk 2]
```diff
@@ -325,6 +325,47 @@ The Dock Manager component raises events when specific end-user interactions are
...
 ```
 
+## Keyboard Navigation
+
+Keyboard navigation in the **Dock Manager** provides a rich variety of keyboard interactions for the user. It enhances the accessibility of the **Dock Manager** and allows to navigate through - documents, tabs and panes. With keyboard navigation the user can traverse over in tabs, which are inside in a [Document host]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcdocumenthost.html), also there is oportunity for spliting the view in multiple directions when there is a [Tab group]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igctabgrouppane.html) with multiple tabs in it. Another key combinations is for docking different type of panes to the [`rootPane`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcdockmanagerlayout.html#rootpane). 
+
+
+Here is a full set of the key combinations:
+
+### Docking Combinations
```
Comment (igdmdimitrov) [inline doc/en/components/dock-manager.md:335]: Split sections to be as following - Docking, Navigating and Other (for close). Also add 'ALT + F6' and 'ALT + SHIFT + F6' combinations.

File: doc/en/components/dock-manager.md [hunk 3]
```diff
@@ -325,6 +325,47 @@ The Dock Manager component raises events when specific end-user interactions are
...
+ - <kbd>Cmd/Ctrl + F7</kbd> Shows the navigator and iterate panes and documents starting from the first document forward
+ - <kbd>Cmd/Ctrl + F8</kbd> Shows the navigator and iterate panes and documents starting from the first document forward
+ - <kbd>Alt + F7</kbd> Shows the navigator and iterate panes and documents starting from the first pane forward
+ - <kbd>Alt + F8</kbd> Shows the navigator and iterate panes and documents starting from the first pane forward
+ - <kbd>Cmd/Ctrl + Shift + F7</kbd> Shows the navigator and iterate panes and documents starting from the last document backwards
+ - <kbd>Cmd/Ctrl + Shift + F8</kbd> Shows the navigator and iterate panes and documents starting from the last document backwards
+ - <kbd>Alt + Shift + F7</kbd> Shows the navigator and iterate panes and documents starting from the last pane backwards
+ - <kbd>Alt + Shift + F8</kbd> Shows the navigator and iterate panes and documents starting from the last pane backwards
+
+Practice all of the above mentioned actions in the demo sample [`demo`](#web-components-dock-manager-overview-example).
```
Comment (igdmdimitrov) [inline doc/en/components/dock-manager.md:367]:
```suggestion
Practice all of the above mentioned actions in the sample [`demo`](#web-components-dock-manager-overview-example). 
```

File: doc/en/components/dock-manager.md [hunk 4]
```diff
@@ -325,6 +325,47 @@ The Dock Manager component raises events when specific end-user interactions are
...
+
+ - <kbd>Shift + Arrow Up</kbd> With multiple tabs in a tab group splits the view and docks the focused tab above
+ - <kbd>Shift + Arrow down</kbd> With multiple tabs in a tab group splits the view and docks the focused tab below
+ - <kbd>Shift + Arrow right</kbd> With multiple tabs in a tab group splits the view and docks the focused tab right
+ - <kbd>Shift + Arrow left</kbd> With multiple tabs in a tab group splits the view and docks the focused tab left
+ - <kbd>Cmd/Ctrl + F6</kbd> Focus next tab in document host
+ - <kbd>Cmd/Ctrl + Arrow right</kbd> Focus next tab in document host
+ - <kbd>Cmd/Ctrl + Shift + F6</kbd> Focus previous tab in document host
+ - <kbd>Cmd/Ctrl + Arrow left</kbd> Focus previous tab in document host
+ - <kbd>Alt + F3</kbd> Closes tab/document
```
Comment (igdmdimitrov) [inline doc/en/components/dock-manager.md:352]:
```suggestion
 - <kbd>Alt + F3</kbd> Closes the active pane
```

File: doc/en/components/dock-manager.md [hunk 5]
```diff
@@ -325,6 +325,47 @@ The Dock Manager component raises events when specific end-user interactions are
 this.dockManager.addEventListener('paneClose', ev => console.log(ev.detail));
 ```
 
+## Keyboard Navigation
+
+Keyboard navigation in the **Dock Manager** provides a rich variety of keyboard interactions for the user. It enhances the accessibility of the **Dock Manager** and allows to navigate through - documents, tabs and panes. With keyboard navigation the user can traverse over in tabs, which are inside in a [Document host]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcdocumenthost.html), also there is oportunity for spliting the view in multiple directions when there is a [Tab group]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igctabgrouppane.html) with multiple tabs in it. Another key combinations is for docking different type of panes to the [`rootPane`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcdockmanagerlayout.html#rootpane). 
+
+
+Here is a full set of the key combinations:
```
Comment (igdmdimitrov) [inline doc/en/components/dock-manager.md:333]:
```suggestion
Keyboard navigation enhances the accessibility of the **Dock Manager** and provides a rich variety of interactions to the end-user - navigate through all panes, split the view in multiple directions through docking the active pane, etc. The shortcuts are as follows:
```

**Comment by zhosti** [comment]
closes #107

File: doc/en/components/dock-manager.md [hunk 6]
```diff
@@ -87,6 +87,12 @@ this.dockManager.layout = {
 };
 ```
 
+With the [`allowFloatingPanesResize`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcdockmanagercomponent.html#allowfloatingpanesresize) property you can choose whether to allow floating panels to be resized, by default the property value is `true`. If the value is set to `false`, none of the floating panes can be resized.
```
Comment (igdmdimitrov) [inline doc/en/components/dock-manager.md:90]:
```suggestion
With the [`allowFloatingPanesResize`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcdockmanagercomponent.html#allowfloatingpanesresize) property you can choose whether to allow floating panes to be resized, by default the property value is `true`. If the value is set to `false`, none of the floating panes can be resized.
```

File: doc/en/components/dock-manager.md [hunk 7]
```diff
@@ -280,7 +287,7 @@ const layout: IgcDockManagerLayout = {
...
+The [`floatingLocation`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcsplitpane.html#floatinglocation), [`floatingWidth`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcsplitpane.html#floatingwidth) and [`floatingHeight`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcsplitpane.html#floatingheight) properties represent absolute dimensions in pixels. With the [`floatingResizable`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcsplitpane.html#floatingresizable) property you can determine whether floating pane resizing is allowed, the default value of the property is `false`. Please note that these properties are applied only for the split panes in the [`floatingPanes`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcdockmanagerlayout.html#floatingpanes) array.
```
Comment (igdmdimitrov) [inline doc/en/components/dock-manager.md:290]:
```suggestion
The [`floatingLocation`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcsplitpane.html#floatinglocation), [`floatingWidth`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcsplitpane.html#floatingwidth) and [`floatingHeight`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcsplitpane.html#floatingheight) properties represent absolute dimensions in pixels. With the [`floatingResizable`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcsplitpane.html#floatingresizable) property you can set whether floating pane resizing is allowed, the default value of the property is `false`. Please note that these properties are applied only for the split panes in the [`floatingPanes`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcdockmanagerlayout.html#floatingpanes) array.
```

**Comment by gedinakova** [comment]
@zhosti "...like navigating through all panes, **spliting** the view in multiple directions through docking the active pane, etc." - should be **splitting**.

**Comment by gedinakova** [comment]
In "Navigating" section, we should probably change 'Focus" to 'Focuses" similar to the other two sections.

**Comment by gedinakova** [comment]
@zhosti "With the allowFloatingPanesResize property you can choose whether to allow floating panels to be resized..." => "whether resizing floating panes is allowed..."

**Comment by gedinakova** [comment]
@zhosti "With the floatingResizable property you can determine whether floating pane resizing is allowed" => "The `floatingResizable`(link) property, which by default is set to true, determines whether a specific floating pane can be resized. This value will overwrite the value of the global `allowFloatingPanesResize` property.

**Comment by gedinakova** [comment]
> @zhosti "With the floatingResizable property you can determine whether floating pane resizing is allowed" => "The `floatingResizable`(link) property, which by default is set to true, determines whether a specific floating pane can be resized. This value will overwrite the value of the global `allowFloatingPanesResize` property.

This comment was about the section in Floating panes, describing the `floatingResizable` property, not the `allowFloatingPanesResize` in Usage.

**Review by HUSSAR-mtrela — APPROVED** [review]

**Review by HUSSAR-mtrela — CHANGES_REQUESTED** [review]
please target vnext branch instead of master branch

**Review by igdmdimitrov — APPROVED** [review]

**Comment by gedinakova** [comment]
@zhosti The demo link does not work (under keyboard navigation). It should be "#web-components-dock-manager-example" instead of "#web-components-dock-manager-overview-example".

**Review by HUSSAR-mtrela — APPROVED** [review]

**Pull request merged**